### DISCUSSION TITLE
My SMP takes a long time to start

### DISCUSSION THREAD

**Discussion by nishiyo** [phax/phoss-smp#281] [General]
Hi,

We are using phoss-smp-webapp-sql.
Recently, SMP containers have been taking longer to start up.
When the number of participants on my SMP increases, it seems to take longer to start.
For example,
 0 participant on my SMP = 20seconds
 [5000 participants] * [some endpoints] on my SMP = 5minutes

I think _performMigrations in SMPMetaManager.java is taking some time.
https://github.com/phax/phoss-smp/blob/2adc695930af895a6cf6085464b4e6af35f80c8e/phoss-smp-backend/src/main/java/com/helger/phoss/smp/domain/SMPMetaManager.java#L153

Does it need to do this process every time the container is started?
In my logs, it takes 5 minutes between "Performing migration 'ensure-transport-profiles-128'" and "Finished performing migration 'ensure-transport-profiles-128'".
(https://github.com/phax/ph-oton/blob/e958adf189b4b4223fea20b2f04344366b5c0fc5/ph-oton-core/src/main/java/com/helger/photon/core/sysmigration/SystemMigrationManager.java#L195)

Do you have any other thoughts about what the cause might be,
and is there any way to avoid this?

thanks.

**Comment by phax** [comment]
Hello @nishiyo, Thanks for reaching out to me.
Usually the "system migration manager" stores the result of its migrations in a file - without checking the code I assume that this manager only writes its results in a file, and that file is not found on restart, because you run the SMP in a containerized environment. And because the file is not found, it executes the migration again.
The solution would be, to store the system migrations in the DB as well.
Please let me know, if my assumptions on your environment are correct - then I will file a bug at https://github.com/phax/ph-oton/issues/

**Reply by nishiyo** [reply]
Yes, I think your assumptions are correct.
We are using "phoss-smp-backend-sql" in a containerized environment.
When container is started, systemmigrations.xml does not exists and "system migration manager" starts migration process.

**Reply by phax** [reply]
Oki. I created https://github.com/phax/ph-oton/issues/20 for that

**Reply by nishiyo** [reply]
Thank you very much!
